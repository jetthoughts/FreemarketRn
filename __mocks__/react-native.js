const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
