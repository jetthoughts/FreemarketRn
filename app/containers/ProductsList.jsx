import React, { Component } from 'react';
import { Text } from 'react-native';

export default class ProductsList extends Component {
  render() {
    return(
      <Text>List of products go here for {this.props.category.name}</Text>
    );
  }
}
