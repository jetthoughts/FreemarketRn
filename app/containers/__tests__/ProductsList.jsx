import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProductsList from '../ProductsList';

it('renders correctly', () => {
  const category = { name: 'Home Decor' };
  const tree = renderer.create(<ProductsList category={category} />).toJSON();

  expect(tree).toMatchSnapshot();
});
