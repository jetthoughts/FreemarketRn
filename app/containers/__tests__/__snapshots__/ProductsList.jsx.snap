// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  List of products go here for 
  Home Decor
</Text>
`;
